romancal/linearity/linearity_step.py

```diff
@@ -0,0 +1,64 @@
+"""
+Apply linearity correction to a science image
+"""
+
+from roman_datamodels import datamodels as rdd
+import numpy as np
+
+from romancal.stpipe import RomanStep
+from romancal.lib import dqflags
+from stcal.linearity.linearity import linearity_correction
+
+__all__ = ["LinearityStep"]
+
+
+class LinearityStep(RomanStep):
+    """
+    LinearityStep: This step performs a correction for non-linear
+    detector response, using the "classic" polynomial method.
+    """
+
+    reference_file_types = ['linearity']
+
+    def process(self, input):
+
+        # Open the input data model
+        with rdd.RampModel(input) as input_model:
```

Comment:
Can you change the input model open to 
       with rdm.open(input) as input_model:
This will allow us to pass the model and not try and open a file.